Adding the building step + CMake Config for Windows

GuillaumeDespaux opened the pull request.

GuillaumeDespaux committed:
Adding the building step

Petite-Brioche0 commented on documents/technical/technical_specification.md:
```diff
@@ -1271,3 +1325,5 @@ To deploy the Quickest Path project, ensure the following environment setup:
 
 [^17]: GNU Compiler Collection: A compiler system produced by the GNU Project.
+
+[^18]: MSYS2: A software distribution and building platform for Windows.
```
is not displayed

Petite-Brioche0 commented on documents/technical/technical_specification.md:
```diff
@@ -964,21 +935,104 @@ The REST API will respond to the user request with the shortest path and the tot
 
 #### 12.3.3. Linux
 
+To do since we don't have the information yet.
```
suggestion : 
We have not yet developed support for the Linux platform. However, full compatibility is planned for a future release.

Petite-Brioche0 reviewed (changes requested):
Last element of the glossary not displayed and the Linux support to be changed

GuillaumeDespaux committed:
Fix issueshttps://github.com/algosup/2024-2025-project-3-quickest-path-team-2/pull/13#discussion_r1941395766

https://github.com/algosup/2024-2025-project-3-quickest-path-team-2/pull/13#discussion_r1941395766
